cypress/integration/device_page_test.js

```diff
@@ -167,15 +174,479 @@ describe('Device page tests', () => {
+              .within(() => {
+                cy.contains(formatBytes(interface.exchanged_bytes));
+                cy.contains(interface.exchanged_msgs);
+              });
+          });
+          const totalBytes = this.deviceDetailed.data.total_received_bytes;
+          const totalMessages = this.deviceDetailed.data.total_received_msgs;
+          const interfacesBytes = _.sumBy(interfaces, 'exchanged_bytes');
+          const interfacesMessages = _.sumBy(interfaces, 'exchanged_msgs');
+          const otherBytes = totalBytes - interfacesBytes;
+          const otherMessages = totalMessages - interfacesMessages;
+          cy.get(`table tbody tr:nth-child(${interfaces.length + 1})`).within(() => {
+            cy.contains('Other');
+            cy.contains(formatBytes(otherBytes));
+            cy.contains(otherMessages);
+          });
+          cy.get(`table tbody tr:nth-child(${interfaces.length + 2})`).within(() => {
+            cy.contains('Total');
+            cy.contains(formatBytes(totalBytes));
+            cy.contains(totalMessages);
+          });
+          cy.get('svg.device-data-piechart').should('be.visible');
+        });
+    });
+
+    it('correctly displays live messages', function () {
+      cy.fixture('config/https').then((config) => {
+        const wssUrl =
+          config.astarte_api_url.replace('https://', 'wss://') + '/appengine/v1/socket/websocket';
+        cy.mockWebSocket({ url: wssUrl });
+        cy.server();
+        cy.route('GET', '/appengine/v1/*/devices/*', this.device);
+        cy.visit(`/devices/${this.device.data.id}`);
+        cy.get('.main-content .card-header')
+          .contains('Device Live Events')
+          .parents('.card')
+          .within(() => {
+            cy.contains(`Joined room for device ${this.device.data.id}`);
+            cy.contains('Watching for device connection events');
+            cy.contains('Watching for device disconnection events');
+            cy.contains('Watching for device error events');
+            cy.contains('Watching for device data events');
+            cy.sendWebSocketDeviceConnected({
+              deviceId: this.device.data.id,
+              deviceIpAddress: '1.2.3.4',
+            });
+            cy.contains('device connected');
+            cy.contains('IP : 1.2.3.4');
+            cy.sendWebSocketDeviceEvent({
+              deviceId: this.device.data.id,
+              event: {
+                interface: 'com.domain.InterfaceName',
+                path: '/some/endpoint',
+                type: 'incoming_data',
+                value: 42,
+              },
+            });
+            cy.contains('incoming data');
+            cy.contains('com.domain.InterfaceName');
+            cy.contains('/some/endpoint');
+            cy.contains('42');
+            cy.sendWebSocketDeviceDisconnected({ deviceId: this.device.data.id });
+            cy.contains('device disconnected');
+            cy.contains('Device disconnected');
```

Comment:
That's because there's both a badge and a message with basically the same words

Reply:
wouldn't it be better to search for a badge containing "Device disconnected"?

Reply:
I tried with `cy.get('.badge').contains('device disconnected')` but `cy.get('.badge')` resolves with other badges before the disconnection badge+message appears. So I left the check like that because it is the simplest check I could think of without using overly specific selectors